dvc/serialize.py

```diff
@@ -39,34 +38,25 @@ def to_dvcfile(stage: "PipelineStage"):
     }
 
 
-def to_lockfile(stage: "PipelineStage") -> OrderedDict:
+def to_lockfile(stage: "PipelineStage") -> dict:
     assert stage.cmd
     assert stage.name
 
-    deps = OrderedDict(
-        [
-            (dep.def_path, dep.remote.get_checksum(dep.path_info),)
-            for dep in stage.deps
-            if dep.remote.get_checksum(dep.path_info)
-        ]
-    )
-    outs = OrderedDict(
-        [
-            (out.def_path, out.remote.get_checksum(out.path_info),)
-            for out in stage.outs
-            if out.remote.get_checksum(out.path_info)
-        ]
-    )
-    return OrderedDict(
-        [
-            (
-                stage.name,
-                OrderedDict(
-                    [("cmd", stage.cmd), ("deps", deps,), ("outs", outs)]
-                ),
-            )
-        ]
-    )
+    res = {"cmd": stage.cmd}
+    deps = [
+        {"path": dep.def_path, dep.checksum_type: dep.get_checksum()}
```

Comment:
This is not as simple for `dvc import`. Maybe let's implement some method in each dep/out class that would return what it thinks it needs to return? Kinda like we did previously.

Reply:
Does this need to support `import`? Or, am I missing something?

Reply:
@skshetry No, it doesn't. It was my bad.